Source/CesiumRuntime/Private/CesiumGltfPrimitiveComponent.cpp

```diff
@@ -20,77 +19,32 @@ PRAGMA_DISABLE_DEPRECATION_WARNINGS
 // Sets default values for this component's properties
 UCesiumGltfPrimitiveComponent::UCesiumGltfPrimitiveComponent() {
   PrimaryComponentTick.bCanEverTick = false;
-  pModel = nullptr;
-  pMeshPrimitive = nullptr;
-  pTilesetActor = nullptr;
+}
+
+UCesiumGltfInstancedComponent::UCesiumGltfInstancedComponent() {
+  PrimaryComponentTick.bCanEverTick = false;
 }
 
 PRAGMA_ENABLE_DEPRECATION_WARNINGS
 
 UCesiumGltfPrimitiveComponent::~UCesiumGltfPrimitiveComponent() {}
 
-void UCesiumGltfPrimitiveComponent::UpdateTransformFromCesium(
-    const glm::dmat4& CesiumToUnrealTransform) {
-  const FTransform transform = FTransform(VecMath::createMatrix(
-      CesiumToUnrealTransform * this->HighPrecisionNodeTransform));
+UCesiumGltfInstancedComponent::~UCesiumGltfInstancedComponent() {}
 
-  if (this->Mobility == EComponentMobility::Movable) {
-    // For movable objects, move the component in the normal way, but don't
-    // generate collisions along the way. Teleporting physics is imperfect, but
-    // it's the best available option.
-    this->SetRelativeTransform(
-        transform,
-        false,
-        nullptr,
-        ETeleportType::TeleportPhysics);
-  } else {
-    // Unreall will yell at us for calling SetRelativeTransform on a static
-    // object, but we still need to adjust (accurately!) for origin rebasing and
-    // georeference changes. It's "ok" to move a static object in this way
-    // because, we assume, the globe and globe-oriented lights, etc. are moving
-    // too, so in a relative sense the object isn't actually moving. This isn't
-    // a perfect assumption, of course.
-    this->SetRelativeTransform_Direct(transform);
-    this->UpdateComponentToWorld();
-    this->MarkRenderTransformDirty();
-    this->SendPhysicsTransform(ETeleportType::ResetPhysics);
-  }
-}
-
-void UCesiumGltfPrimitiveComponent::BeginDestroy() {
+namespace {
+void destroyCesiumPrimitive(UStaticMeshComponent* uobject) {
```

Comment:
We're a little inconsistent about naming in Cesium for Unreal. Sometimes we use the cesium-native conventions (local variable and parameter names start with a lowercase letter, pointers prefixed with `p`), and sometimes we use the Unreal Engine conventions (capital first letter, no `p`). The latter is important for the bits that are public, because following Unreal's standard makes it easier for people to use our plugin. The former I guess we just do where it doesn't matter out of habit from working in cesium-native. But `uobject` is kind of a third way. At the risk of nitpicking, I think I'd rename this to either `pComponent` or `Component`.

Reply:
I did change this, but I'm curious: does the "pComponent" convention extend to things that behave like pointers, like std::shared_ptr or std::unique_ptr?

Reply:
> I'm curious: does the "pComponent" convention extend to things that behave like pointers, like std::shared_ptr or std::unique_ptr?

Yes, smart pointers count as pointers. An explanation of why we use the `p` prefix is in the [style guide](https://github.com/CesiumGS/cesium-native/blob/main/doc/style-guide.md):

> DON'T use Hungarian notation, except that it's useful to prefix pointer-like variables with a p, e.g. pThing. This is justified because it's not meant to convey type information (not so useful), but rather as an easily-understood abbreviation for the descriptive name of what the variable actually is (very useful). The variable isn't a thing, it's a pointer to a thing, but pointerToThing would get annoying quickly. (see [NL.5](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nl5-avoid-encoding-type-information-in-names))

Although not called out in the style guide explicitly, this extends to other similar cases as well. For example, we often prefix the name of variables of std::optional with `maybe`. Once again, this isn't Hungarian notation, it's a description of what the variable actually is. If an `int32_t` is an index, a `std::optional<int32t_t>` could be an `optionalIndex` or `maybeAnIndex` or, our usual convention, `maybeIndex`.